src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -0,0 +1,215 @@
+/*-------------------------------------------------------------------------
+ *
+ * insert_select_planner.c
```

Comment:
I think that's a worthwhile change, but it would blow up the PR quite a bit. Would you be ok with doing that as a separate PR?